[Windows/Bug]Guide/Home button on Xinput controller bug with explorer.exe terminated

Hi,

I have this exact same issue which is bothering me : https://www.reddit.com/r/RetroArch/comments/ensw3o/i_think_i_discovered_a_bug_with_retroarch_anyone/

Pressing the Guide button when explorer.exe is off make Retroarch start a server, it then becomes extremely slow and I have to restart it

Just tried out of curiosity and can confirm this oO 

> [ERROR] [dispserv]: HrInit of ITaskbarList3 failed.
...
[INFO] [netplay] Waiting for client ...
[INFO] [CONTENT LOAD]: CRC32: 0x3337ec46 .
[INFO] [netplay] You have joined as player 1
[INFO] [netplay] Port Mapping Successful: xx.xx.xx.xx:55435
[INFO] [DISCORD]: Netplay room details: ID=78806, Nick=bslenul IP=xx.xx.xx.xx Port=55435
[INFO] [DISCORD]: Join secret: 78806|xxxxxxxxxx
[INFO] [DISCORD]: Party ID: 78806

Not sure what the "Join secret" is, so I preferred to hide the 2nd part of that line.

If no content is loaded you get that notification: `Netplay will start when content is loaded`. Steam completely closed, I even closed DS4Windows just in case... It's like it's pressing the "Netplay Hosting" hotkey, except it's not bound to anything for me 😅

--- a/input/drivers/dinput.c
+++ b/input/drivers/dinput.c
@@ -265,8 +265,13 @@ static void dinput_poll(void *data)
          }
       }
       else
+      {
          /* Shifts only when window focused */
          dinput_keyboard_mods(di, RETROKMOD_SHIFT);
+
+         /* Ignore 'unknown/undefined' key */
+         di->state[RETROK_UNKNOWN] = 0;
+      }
 
       /* Left alt keyup when unfocused, to prevent alt-tab sticky */
       dinput_keyboard_mods(di, RETROKMOD_ALT);

--- a/input/drivers/winraw_input.c
+++ b/input/drivers/winraw_input.c
@@ -500,6 +500,7 @@ static LRESULT CALLBACK winraw_callback(
          /* Ignored scancodes */
          switch (mcode)
          {
+            case RETROK_UNKNOWN:
             case 0xE11D:
             case 0xE02A:
             case 0xE036:

